external/fv3fit/fv3fit/keras/_models/models.py

```diff
@@ -44,13 +44,16 @@
 
 @register_training_function("DenseModel", DenseHyperparameters)
 def train_dense_model(
-    input_variables: Iterable[str],
-    output_variables: Iterable[str],
     hyperparameters: DenseHyperparameters,
     train_batches: Sequence[xr.Dataset],
     validation_batches: Sequence[xr.Dataset],
 ):
-    model = DenseModel("sample", input_variables, output_variables, hyperparameters)
+    model = DenseModel(
+        "sample",
+        hyperparameters.input_variables,
```

Comment:
I know it would involve changes across a lot of files for the models and training functions, but should we eventually get rid of the input/output variable args since they should always be in the hyperparameters after this PR?

Reply:
I considered doing so in this PR, but Predictor specifies that the init should take in input and output variables as arguments. This is somewhat reasonable, since Predictor needs to report what the input and output variables are as properties. The "issue" here leading to weirdness is that DenseModel is both our training code and our Predictor (which is sort of why we moved to training functions).

I think it shouldn't be too much of an issue to leave this as the case for the Dense and RandomForest model training routines. If we ever get around to refactoring the objects into functions this problem will go away, but there's no pressing need to do so.